### ISSUE TITLE
Duplicated struct not identified at compile time

### ISSUE THREAD

**Issue by yloiseau** [eclipse/golo-lang#323]
When a module contains two identically named structures, as in
```golo
module Test

struct Toto = { x }
  
struct Toto = { x, y }

function main = |args| {
  let t2 = Toto(1, 2)
}
```
the code compile without complaining, but fail at runtime, with `java.lang.ClassFormatError: Duplicate method name&signature` when `golo run` and with  `java.lang.LinkageError: loader (instance of GoloClassLoader): attempted duplicate class definition` when `golo golo`.

*Intended behavior*: should fail at *compile time* with explicit error, e.g.
```
[err] Duplicated `struct` definition  for "Toto" at line 5, column 1
```

**Comment by yloiseau** [comment]
Same thing for duplicated `union` and `function`, that fail at runtime but compile without error

**Comment by jponge** [comment]
:+1:

**Comment by yloiseau** [comment]
The `function` case was fixed in #330
